hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ImmutableSegment.java

```diff
@@ -43,4 +48,127 @@ public KeyValueScanner getKeyValueScanner() {
     return new CollectionBackedScanner(getCellSet(), getComparator());
   }
 
+  public boolean isFlat() {
```

Comment:
make this private method so that only the segment knows that it is flat and applies flattening in this case